src/coap_address.c

```diff
@@ -876,12 +951,24 @@ coap_resolve_address_info(const coap_str_const_t *address,
     }
   }
   return info_list;
+}
+
 #else
-#bad OS type not supported
+
+coap_addr_info_t *
+coap_resolve_address_info(const coap_str_const_t *address,
+                          uint16_t port,
+                          uint16_t secure_port,
+                          uint16_t ws_port,
+                          uint16_t ws_secure_port,
+                          int ai_hints_flags,
+                          int scheme_hint_bits,
+                          coap_resolve_type_t type) {
   return NULL;
-#endif
 }
 
+#endif
```

Comment:
I think I prefer the #bad to be here so a new platform will throw up a compile warning rather that having to debug an address that is not parsing.